### PR TITLE
feat: Adds the ability to edit and delete custom attributes

### LINKED ISSUE #2865: feat: Add the ability to edit/delete custom attributes
- Upon clicking the edit/delete button in the attribute table. The user should be able to edit/delete the attributes

**Note**
- The `attribute_type` and `attribute_model` are not editable

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

Adds the ability to edit and delete custom attributes that have been created.

Fixes #2865 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Can be tested by going to the Setting page and select Attributes.

**Screen recording**

https://user-images.githubusercontent.com/64252451/132481967-098f55f5-b1cc-495c-8fd5-b4c45707217a.mov

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by fayazara — DISMISSED** [review]

File: app/javascript/dashboard/routes/dashboard/settings/attributes/CustomAttribute.vue [hunk 1]
```diff
@@ -123,6 +170,33 @@ export default {
     fetchAttributes(index) {
       this.$store.dispatch('attributes/get', index);
     },
+    async deleteAttributes({ id }) {
+      try {
+        await this.$store.dispatch('attributes/delete', id);
+        this.showAlert(this.$t('ATTRIBUTES_MGMT.DELETE.API.SUCCESS_MESSAGE'));
+      } catch (error) {
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/settings/attributes/CustomAttribute.vue:177]: Check any message exist in the `error` object. Otherways show the default error message

File: app/javascript/dashboard/routes/dashboard/settings/attributes/EditAttribute.vue [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            {{ $t('ATTRIBUTES_MGMT.ADD.FORM.KEY.LABEL') }}
+            <i class="ion-help" />
+          </label>
+          <p class="key-value text-truncate">
+            {{ attributeKey }}
+          </p>
+        </div>
+      </div>
+      <div class="modal-footer">
+        <woot-button :is-loading="uiFlags.isCreating">
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/settings/attributes/EditAttribute.vue:66]: Check `isCreating` is using  or not

File: app/javascript/dashboard/routes/dashboard/settings/attributes/EditAttribute.vue [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        {
+          id: 0,
+          option: 'Conversation',
+        },
+        {
+          id: 1,
+          option: 'Contact',
+        },
+      ],
+      types: [
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/settings/attributes/EditAttribute.vue:106]: The constants `models` and `types` are used in both AddAtrribute and EditAttribute components. Let's move this into constants.

File: app/javascript/dashboard/routes/dashboard/settings/attributes/EditAttribute.vue [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    ...mapGetters({
+      uiFlags: 'attributes/getUIFlags',
+    }),
+    pageTitle() {
+      return `${this.$t('ATTRIBUTES_MGMT.EDIT.TITLE')} - ${
+        this.selectedAttribute.attribute_display_name
+      }`;
+    },
+    attributeKey() {
+      return this.displayName
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/settings/attributes/EditAttribute.vue:160]: Move to helper methods

File: app/javascript/dashboard/routes/dashboard/settings/attributes/EditAttribute.vue [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+          attribute_display_name: this.displayName,
+          attribute_description: this.description,
+          attribute_model: this.attributeModel,
+          attribute_display_type: this.attributeType,
+          attribute_key: this.attributeKey,
+        });
+        this.showAlert(this.$t('ATTRIBUTES_MGMT.EDIT.API.SUCCESS_MESSAGE'));
+        this.onClose();
+      } catch (error) {
+        this.showAlert(this.$t('ATTRIBUTES_MGMT.EDIT.API.ERROR_MESSAGE'));
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/settings/attributes/EditAttribute.vue:194]: Check any message exist in the `error` object. Otherways show the default error message

**Comment by muhsin-k** [comment]
@iamsivin Nice work. Please see the inline comments.

@pranavrajs Do we need to give permission to edit attribute type and attribute model?

**Comment by pranavrajs** [comment]
@muhsin-k @iamsivin I don't think we need to provide a option to edit the attribute model since it means that we are moving the contact attribute to conversation attribute.

I guess the type is okay as far as there are no values created. Otherwise, if the type is being changed, we need reset the values already assigned to a conversation or contact.

File: app/javascript/dashboard/routes/dashboard/settings/attributes/attributeHelper.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export const convertToAttributeKey = name => {
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/settings/attributes/attributeHelper.js:1]: Add specs

**Review by muhsin-k — APPROVED** [review]
Please disable the attribute route before merging the PR

**Review by muhsin-k — APPROVED** [review]

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**